src/index.ts

```diff
@@ -8,7 +8,7 @@ import React, {
 } from "react"
 import { unionWith, eqBy, prop, clone } from "ramda"
 import _ from "lodash"
-import { History as HHistory } from "history"
+import type { History as HHistory } from "history"
```

Comment:
`import type` statements are removed when Typescript is compiled to Javascript.

A consuming typescript project that uses `history` v5 will use types directly from `history`, which provides its own types starting in v5.

A consuming typescript project that uses `history` v4 will use types from `@types/history`.

A consuming javascript project will not use types.